Library/ViewModels/LoginViewModel.swift

```diff
@@ -171,7 +174,10 @@ public final class LoginViewModel: LoginViewModelType, LoginViewModelInputs, Log
     self.logIntoEnvironment
       .observeValues { _ in AppEnvironment.current.koala.trackLoginSuccess(authType: Koala.AuthType.email) }
 
-    self.showHidePasswordButtonToggled = self.shouldShowPasswordProperty.signal
+    self.showHidePasswordButtonToggled = Signal.combineLatest(
```

Comment:
In this case let's consider what we're wanting to express with the `Signal`. We're saying:

1. Emit `showHidePasswordButtonToggled` when `shouldShowPasswordProperty` emits its value.
1. _Also_ emit that same value when `traitCollectionDidChange` emits.

So by using `comineLatest` here, we're requiring that both of the signals emit in order for `showHidePasswordButtonToggled` to emit, which isn't really necessary. Instead I think what we want is to `merge` like this:

```swift
let shouldShowPassword = self.shouldShowPasswordProperty.signal

self.showHidePasswordButtonToggled = Signal.merge(
  shouldShowPassword,
  shouldShowPassword.takeWhen(self.traitCollectionDidChangeProperty.signal)
)
```

This captures it a little more clearly by saying that we want to _take_ the value of `shouldShowPassword` _when_ `self.traitCollectionDidChangeProperty.signal` emits.